jykuo-love-shiritori/twp
# [frontend] search

Resolveeeeeeeees part of #94

> [!IMPORTANT]  
> #132 is not totally resolve yet !

### Changes & Testing Approach


### (L) Search TYK ⌛
  - test in global page with `desktop` or `phone`
    - login then go to index page or whatever page excepts for shop page
    - search whatever you want in search bar then `enter` or click the `search button`
    - then you will find you in the search page with filter
  - test in search page with filter
    - input negative nums in price or stock field --> should alert
    - input folat nums in price or stock field --> should alert
    - input min but not max(vice versa) at the same time with price or stock field --> should alert
    -  input NaN in price or stock field --> should alert
    - set min > max --> should alert
    - change the text in search bar, the url and result will change
    - reload should keep the status
  - EXAMPLE test global page (please check url as well!)
    - input `pr` in searchbar then enter, you will find 5 items
    - click `has coupon` and click `submit`, you will find 4 items with pr
    - input 1 in min price and 100 in max price, you will find 3 items with pr
    - reload, all attributes should keep on webpage
- [ ] /search/shop
  - i don't do this because i think there is no way(page) can trigger it and i think the current status is enough
- [x]   /shop/{seller_name}/search
  - test in shop page with `desktop` or `phone`
    - login then go to shop page
    - search whatever you want in search bar then `enter` or click the `search button`
    - then you will find you in the search page `within shop page` with filter
  - test in search page with filter
    - input negative nums in price or stock field --> should alert
    - input folat nums in price or stock field --> should alert
    - input min but not max(vice versa) at the same time with price or stock field --> should alert
    -  input NaN in price or stock field --> should alert
    -  set min > max --> should alert
    - change the text in search bar, the url and result will change
    - reload should keep the status
  - EXAMPLE test shop page (please check url as well!)
    - input `1` in searchbar then enter, you will find 1 item
    - in the same page, input `2` in searchbar then enter, you will find 1 item
    
### Test Environment Setup
open 3 terminals
one for `docker-compose up postgres minio `
one for `go run .`
one for `cd frontend` -> `npm i` -> `npm run build`
then open the browser then enter http://localhost:8080/ 

### Additional context


#132 is not resolve yet
